### PR TITLE
Encode string

### PR DESCRIPTION
**Pull request by colin99d**
# Description

@martinb-bb right now this is a bad implementation because it uses increasing numbers, so the computer believe there are relationships that do not actually exist. Lets talk about a better implementation.

# How has this been tested?

* Please describe the tests that you ran to verify your changes. 
* Provide instructions so we can reproduce. 
* Please also list any relevant details for your test configuration.
- [ ] Make sure affected commands still run in terminal
- [ ] Ensure the api still works
- [ ] Check any related reports

# Checklist:

- [ ] Update [our Hugo documentation](https://openbb-finance.github.io/OpenBBTerminal/) following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/website).
- [ ] Update our tests following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/tests).
- [ ] Make sure you are following our [CONTRIBUTING guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/blob/main/CONTRIBUTING.md).
- [ ] If a feature was added make sure to add it to the corresponding [scripts file](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/scripts).

# Others
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.

### CODE REVIEW

File: openbb_terminal/forecast/forecast_model.py [hunk 1]
```diff
@@ -542,3 +542,24 @@ def corr_df(df: pd.DataFrame) -> pd.DataFrame:
...
+        The dataset you wish to modify
+    target_column : str
+        The column you wish to encode
+
+    Returns
+    -------
+    pd.DataFrame:
+        Dataframe with added encoded column
+    """
+    dataset[target_column] = dataset[target_column].astype("category").cat.codes
```
Comment (jmaslek) [inline openbb_terminal/forecast/forecast_model.py:563]: Why not the standard sklearn.preprossessing.LabelEncoder?

**Comment by jmaslek** [comment]
This is in general a bad idea because as you say, it associates higher numbers.  Look into one hot encoding (which can possibly add a lot of variables) but is better for non-ordinal features

**Comment by martinb-bb** [comment]
@colin99d Something to look into is to see if Darts supports non number variables. Might be a simple solution.

File: openbb_terminal/forecast/forecast_model.py (hunk 1 above)
Comment (DidierRLopes) [reply openbb_terminal/forecast/forecast_model.py:563]: Yup, that's why Colin added that description. We already talked about this 👍🏽 

We should defo be using sklearn for this.

I would still allow to have an option for this type of encoding which makes sense in the context of quantitative variables, e.g. "How hot was the water": "cold", "not much". "hot", "boiling". In here 1,2,3,4 would make sense.

Comment (martinb-bb) [reply openbb_terminal/forecast/forecast_model.py:563]: For reference: 

```python
>>> from sklearn import preprocessing
>>> le = preprocessing.LabelEncoder()
>>> le.fit(["paris", "paris", "tokyo", "amsterdam"])
LabelEncoder()
>>> list(le.classes_)
['amsterdam', 'paris', 'tokyo']
>>> le.transform(["tokyo", "tokyo", "paris"])
array([2, 2, 1]...)
```
Then we can replace the column with this newly encoded array?

Comment (martinb-bb) [reply openbb_terminal/forecast/forecast_model.py:563]: Also, for later versions: https://stackoverflow.com/questions/24458645/label-encoding-across-multiple-columns-in-scikit-learn

**Comment by martinb-bb** [comment]
Also, I think we may not want to make this is a menu function. Instead just have it as an automatic check and process when creating `TimeSeries()` and the user specifies a categorical column as a covariate or forecast target.

File: openbb_terminal/forecast/forecast_model.py (hunk 1 above)
Comment (jmaslek) [reply openbb_terminal/forecast/forecast_model.py:563]: > Yup, that's why Colin added that description. We already talked about this 👍🏽
> 
> We should defo be using sklearn for this.
> 
> I would still allow to have an option for this type of encoding which makes sense in the context of quantitative variables, e.g. "How hot was the water": "cold", "not much". "hot", "boiling". In here 1,2,3,4 would make sense.

Yes, but if the encoder sees cold, boiling, hot, not much, in that order, it will encode it as 1,2,3,4 even though that is not the right ordinal ordering.  What kind of strings are being encoded?

Comment (DidierRLopes) [reply openbb_terminal/forecast/forecast_model.py:563]: Yup. But there are examples where preserving that relationship between variables can be important. We could allow the user to specify those :)  But prob would be better to have this as a request rather than just adding it without knowing if users would like that.

**Comment by colin99d** [comment]
We are holding off on this for now. Darts static covariates might handle this already.
